server/streaming/server.go

```diff
@@ -127,16 +134,72 @@ func (s *Server) ServeBinaryWs(config *config.Config) func(w http.ResponseWriter
 				s.logger.ErrorLog("extract_sender_id_err", err, nil)
 			}
 
+			binarySerializer := telemetry.NewBinarySerializer(requestIdentity, s.DispatchRules, s.logger)
 			socketManager := NewSocketManager(ctx, requestIdentity, ws, config, s.logger)
-			s.registry.RegisterSocket(socketManager)
-			defer s.registry.DeregisterSocket(socketManager)
+			s.registerSocket(socketManager, binarySerializer)
+			defer s.deregisterSocket(socketManager, binarySerializer)
 
-			binarySerializer := telemetry.NewBinarySerializer(requestIdentity, s.DispatchRules, s.logger)
 			socketManager.ProcessTelemetry(binarySerializer)
 		}
 	}
 }
 
+func (s *Server) dispatchConnectivityEvent(sm *SocketManager, serializer *telemetry.BinarySerializer, event protos.ConnectivityEvent) error {
+	connectivityDispatcher, ok := s.DispatchRules[connectitivityTopic]
+	if !ok {
+		return nil
+	}
+
+	connectivityMessage := &protos.VehicleConnectivity{
+		Vin:       sm.requestIdentity.DeviceID,
+		CreatedAt: timestamppb.Now(),
+		Status:    event,
+	}
+
+	payload, err := proto.Marshal(connectivityMessage)
+	if err != nil {
+		return nil
+	}
+
+	streamMessage := messages.StreamMessage{
+		TXID:         []byte(fmt.Sprintf("%s-%s", connectitivityTopic, uuid.NewString())),
+		SenderID:     []byte(sm.requestIdentity.SenderID),
+		DeviceID:     []byte(sm.requestIdentity.DeviceID),
+		DeviceType:   []byte("vehicle_device"),
+		MessageTopic: []byte(connectitivityTopic),
+		Payload:      payload,
+		EnvMessageID: []byte(uuid.NewString()),
+		CreatedAt:    uint32(connectivityMessage.CreatedAt.AsTime().Unix()),
+	}
```

Comment:
`telemetry.NewRecord` requires flatbuffer to build a record object. We might need to do a big refactor to make it accept a non flatfbuffer message so creating this intermediate object